x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ml/job/process/autodetect/output/FlushAcknowledgement.java

```diff
@@ -31,39 +32,54 @@
     public static final ParseField LAST_FINALIZED_BUCKET_END = new ParseField("last_finalized_bucket_end");
 
     public static final ConstructingObjectParser<FlushAcknowledgement, Void> PARSER = new ConstructingObjectParser<>(
-            TYPE.getPreferredName(), a -> new FlushAcknowledgement((String) a[0], (Date) a[1]));
+            TYPE.getPreferredName(), a -> new FlushAcknowledgement((String) a[0], (Instant) a[1]));
 
     static {
         PARSER.declareString(ConstructingObjectParser.constructorArg(), ID);
         PARSER.declareField(ConstructingObjectParser.optionalConstructorArg(),
-                p -> TimeUtils.parseTimeField(p, LAST_FINALIZED_BUCKET_END.getPreferredName()),
+                p -> TimeUtils.parseTimeFieldToInstant(p, LAST_FINALIZED_BUCKET_END.getPreferredName()),
                 LAST_FINALIZED_BUCKET_END, ObjectParser.ValueType.VALUE);
     }
 
-    private String id;
-    private Date lastFinalizedBucketEnd;
+    private final String id;
+    private final Instant lastFinalizedBucketEnd;
 
-    public FlushAcknowledgement(String id, Date lastFinalizedBucketEnd) {
+    public FlushAcknowledgement(String id, Instant lastFinalizedBucketEnd) {
         this.id = id;
-        this.lastFinalizedBucketEnd = lastFinalizedBucketEnd;
+        // The C++ passes 0 when last finalized bucket end is not available
+        long epochMillis = (lastFinalizedBucketEnd != null) ? lastFinalizedBucketEnd.toEpochMilli() : 0;
```

Comment:
Could you re-use the same pattern as in `FlushJobAction.Response` here?
```
            this.lastFinalizedBucketEnd =
                (lastFinalizedBucketEnd != null) ? Instant.ofEpochMilli(lastFinalizedBucketEnd.toEpochMilli()) : null;
```

Reply:
It's slightly different as this one can be parsing output from the C++, which writes 0 instead of null.  So this one needs to convert 0 to null.